### ISSUE TITLE
Improve error message when track / problem cannot be identified

### ISSUE THREAD

**Issue by kytrinyx** [exercism/cli#197]
A lot of people have trouble with the submit command, because they've put their solution files outside of the exercism working directory.

https://github.com/exercism/exercism.io/issues/2335

Perhaps we can improve the error message to explain what was expected as opposed to what actually happened ("you gave me this... but I'm looking for something like this.").

**Comment by anxiousmodernman** [comment]
Got your email about 24 pull requests. 'Tis the season! :gift: 

Here is a draft of a more verbose error message this situation based on my paths.

```
2015/11/29 22:16:58 Unable to submit - Unable to identify track and file.

This usually occurs when you try to submit a file located outside of your configured exercises directory.

Your current directory is /Users/coleman/random-directory; exercism is looking for your exercises directory at /Users/coleman/exercism.

Try running `exercism submit <filename>` from inside /Users/coleman/exercism/<language>/<problem>.

For example, to submit the JavaScript "hello-world.js" problem, run `exercism submit hello-world.js` from the directory /Users/coleman/exercism/javascript
```

This is verbose, and maybe it's overkill, but based on [exercism/exercism.io#2335](https://github.com/exercism/exercism.io/issues/2335) it seems like more info is better than less.

To get the nice output for all OS's, we will probably need string templating + a helper function, so we could do something like this in `api/iteration.go`

```go

	// All the files should be within the exercism path.
	for _, filename := range filenames {
		if !iter.isValidFilepath(filename) {
			msg := getUnidentifiableTrackMessage(dir)
			return nil, errors.New(msg)
		}
	}
```

If anyone has comments, let me know. An alternative to this message might be introducing some kind of a subcommand/flag for `exercism submit`, so an error might instead look like

```
2015/11/29 22:16:58 Unable to submit - Unable to identify track and file.
Please run `exercism submit --help` for more info
```

**Comment by kytrinyx** [comment]
I think you're right that more information is better than less. If they're getting the error, then we may as well show them the error message up front rather than making them re-run the command.

We can know for sure whether or not the problem is that they're outside of their configured directory, since we have both paths. If that's not the problem, let's show a more generic error and ask them to submit an issue.

What are the suggested ways to fix this?

1. Change their configuration. Can we guess what it should be and suggest that?
2. Move their code to the correct place. This involves understanding the DIR/track_id/slug thing.

A couple of things. Would using whitespace to show the diff between current/configured help?

```
Configured:  /Users/coleman/exercism
Expected:    /Users/coleman/exercism/<TRACK_ID>/<EXERCISE_SLUG>/
Current:        /Users/coleman/random-directory
```

We could tell them how to change the configuration, but we don't know where they have their code, and guessing seems perilous in this case. It would be a shame to give them a command to run and have it be wrong, since that would just be more confusing.

> For example, to submit the JavaScript "hello-world.js" problem, run `exercism submit hello-world.js` from the directory /Users/coleman/exercism/javascript

Actually, that would be `/Users/coleman/exercism/javascript/hello-world`. I wonder if the easiest thing is to tell them to fetch again, move their solution to where the CLI put the new code, and then submit from there.

**Comment by benjic** [comment]
This issue should probably be closed since it was implemented in #261

**Comment by anxiousmodernman** [comment]
Agreed. I probably should have referenced it.
